src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,240 @@ class Guides::Database::Querying < GuideAction
-    * `created_at` (automatically added)
-    * `updated_at` (automatically added)
-    * `id` (automatically added)
-    * `age`
-    * `name`
+    When you run any query, you'll get back an instance of the model, array of that model instance,
+    nil, or raise an exception (e.g. `Avram::RecordNotFoundError`).
 
-
-    ## Distinct on
+    For our examples, we will use this `User` model.
 
     ```crystal
-    UserQuery.new.distinct_on(&.name)
+    class User < BaseModel
+      table :users do
+        # `id`, `created_at`, `updated_at` predefined for us
+        column name : String
+        column age : Int32
+        column admin : Bool
+      end
+    end
     ```
 
-    ## Where clauses
+    ### Simple helper methods
 
-    You can use the methods generated by Lucky to create where clauses.
+    By default, all query objects include `Enumerable(T)` methods like `each`, and `map`.
+    There's also a few class methods for doing quick finds.
 
-    ```crystal
-    # SELECT * FROM users where name='Paul';
-    UserQuery.new.name("Paul")
+    * `first` - Returns the first record. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `first?` - Returns the first record. Returns `nil` if no record is found.
+    * `find(id)` - Returns the record with the primary key `id`. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `last` - Returns the last record. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `last?` - Returns the last record. Returns `nil` if no record is found.
+    * `all` - Returns an array of all the records.
 
-    # SELECT * FROM users where age=28;
-    UserQuery.new.age(28)
-    ```
+    ## Simple Selects
 
-    ### Parsing query params
+    > When doing a `SELECT`, Avram will select all of the columns individually (i.e. `users.id,
+    > users.created_at, users.updated_at, users.name, users.age, users.admin`) as opposed to `*`.
+    > However, for brevity, we will use `COLUMNS`.
 
-    Lucky will accept the type defined in the model (e.g. `Int32` for `age`) or a
-    `String`. If you pass it a `String` then Lucky will attempt to parse it and use
-    the type defined for the column.
+    ### Select all
 
-    For example:
+    `SELECT COLUMNS FROM users`
 
     ```crystal
-    # Lucky parses the string and use an Int32 in the query
-    UserQuery.new.age("30")
-
-    # Won't accept other types
-    # Fails at compile time because you can't pass a Float64
-    UserQuery.new.age(30.0)
+    users = UserQuery.all
     ```
 
-    > Note that this will change in the future but this is the current behavior. In
-    the future only the defined type (`Int32` in this example) will work. You will
-    then need to manually parse it. Don't worry though, Lucky will add some helpers
-    to actions and the model to make this easy and maintainable.
-
-    ## Ordering results
-
-    ````crystal
-    UserQuery.new.age.desc_order
-    # or
-    UserQuery.new.age.asc_order
-    ````
+    ### Select one
```

Comment:
I think splitting these into `Select first`, `select last` and `select by primary key (find)`. That way they will show up in search better (the heading will be what people search for. May also be worth noting there are `?` methods of these and that if you use the non-question mark version it will raise if it doesn't exist.  Using `?` will return `nil` if it doesn't exist

You do this above but I think a note here would be good as well in case they get here by a search and don't see the previous sections